modules/sql-engine/src/test/java/org/apache/ignite/internal/sql/engine/framework/TestNode.java

```diff
@@ -203,7 +200,7 @@ public QueryPlan prepare(String query) {
     public QueryPlan prepare(SqlNode queryAst) {
         assertThat(queryAst, not(instanceOf(SqlNodeList.class)));
 
-        return await(prepareService.prepareAsync(queryAst, createContext()));
+        return await(((PrepareServiceImpl) prepareService).prepareAsync(queryAst, createContext()));
```

Comment:
once again: TestNode _must_ (I mean even cast like this is no allowed) use interface, not implementation

Reply:
Benchmarks need a way to prepare plan from AST, to avoid parsing stage from benchmark.
There are next ways to fix this:
* get prepare service from node (e.g. via 'unwrap' method), then cast to specific implementation
* pull `prepareAsync` method into the interface
* introduce the interface(s) with separate methods for parsing and optimization stages, introduce handler objects for these interface(s), and pass then to PrepareService constructor. Then either call `unwrap` method on prepare service to get these handlers.
* Rework benchmark to make it uses PrepareServiceImpl directly.